### PR TITLE
Action ingestion project tagging

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description

Adds RBAC project tagging to the Chef Action ingestion.

### :athletic_shoe: Demo Script / Repro Steps

1. `build components/ingest-service`
1. `start_all_services`
1. `send_chef_action_example`
1. `curl $ELASTICSEARCH_URL/actions-*/_search | jq . | less`
1. Ensure 'some project name' is in the project array

### :chains: Related Resources

issue - https://github.com/chef/automate/issues/62
similar PR but for CCRs - https://github.com/chef/a2/pull/5236

### :white_check_mark: Checklist

- [x] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [x] Code actually executed?
- [x] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/ingest-service/pipeline/processor/action_project_tagger_test.go [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+	<-out
+
+	assert.Equal(t, 1, listProjectRulesCount)
+}
+
+// When 5 messages are in the inbox the ListProjectRules function is only called once.
+func TestActionBundler5Messages(t *testing.T) {
+	inbox := make(chan message.ChefAction, 100)
+	listProjectRulesCount := 0
```
Comment (phiggins) [inline components/ingest-service/pipeline/processor/action_project_tagger_test.go]: I'm not familiar with gomock at all, but from using other mocking libraries I think this is what the `Times` function is for: https://godoc.org/github.com/golang/mock/gomock#Call.Times

File: components/ingest-service/pipeline/processor/action_project_tagger_test.go [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+	inbox <- action2
+	close(inbox)
+
+	out := BuildActionProjectTagger(authzClient)(inbox)
+
+	processMsg1 := <-out
+	assert.Equal(t, []string{}, processMsg1.InternalChefAction.Projects)
+
+	processMsg2 := <-out
+	assert.Equal(t, []string{}, processMsg2.InternalChefAction.Projects)
```
Comment (phiggins) [inline components/ingest-service/pipeline/processor/action_project_tagger_test.go]: I'm finding these tests a bit hard to understand, and I wonder if the tests wouldn't need to do as much if the logic for how to treat the messages and rules was separated from the channel handling /message passing code.

File: components/ingest-service/pipeline/processor/action_project_tagger_test.go (hunk 1 above)
Comment (lancewf) [reply components/ingest-service/pipeline/processor/action_project_tagger_test.go]: Thanks. I never thought of doing that. 
https://github.com/chef/automate/pull/214/commits/c1537a72e4990bfe1d952b044706ff0708d596c2

File: components/ingest-service/pipeline/processor/action_project_tagger_test.go (hunk 2 above)
Comment (lancewf) [reply components/ingest-service/pipeline/processor/action_project_tagger_test.go]: 🤔 I will look into this.

Comment (lancewf) [reply components/ingest-service/pipeline/processor/action_project_tagger_test.go]: I think this last test (TestActionBundlerMatchProjectRuleNodeRuleType) is not needed, so I removed it. This is being tested here https://github.com/chef/automate/pull/214/files/ca2de00b00ff7b701f275efcf0cbcfea39652b8c#diff-7e0662f4628cc26ca7731861f082208fR63.

File: components/ingest-service/pipeline/processor/action_project_tagger_test.go [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+	<-out
+	<-out
+	<-out
+	<-out
+	<-out
+}
+
+// A simple run through of the bundle project tagger processor.
+// Two messages are sent through with only one matching a project.
+func TestActionBundlerMatchProjectRule(t *testing.T) {
```
Comment (lancewf) [inline components/ingest-service/pipeline/processor/action_project_tagger_test.go]: This has mostly been tested with this test here https://github.com/chef/automate/pull/214/files#diff-7e0662f4628cc26ca7731861f082208fR20, but I wanted to do a whole run through of the entire processor.

File: components/ingest-service/pipeline/processor/action_project_tagger_test.go (hunk 2 above)
Comment (lancewf) [reply components/ingest-service/pipeline/processor/action_project_tagger_test.go]: I went through and added more comments on what is going on in these tests. I do not know a way to break apart the channel handling from the bundler algorithm. This algorithm was designed to use the length of the channel to figure out how many messages to bundle together. Then the algorithm counts down each message as it goes through.

**Pull request merged**